syncer/peers/nofork/manager.go

```diff
@@ -0,0 +1,241 @@
+package nofork
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/0xPolygon/polygon-edge/blockchain"
+	"github.com/0xPolygon/polygon-edge/network/grpc"
+	"github.com/0xPolygon/polygon-edge/syncer/peers"
+	"github.com/0xPolygon/polygon-edge/syncer/peers/nofork/proto"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+const (
+	SyncPeersManagerLoggerName = "sync-peers-manager"
+)
+
+type SyncPeersManager struct {
+	logger     hclog.Logger
+	network    Network
+	blockchain Blockchain
+	server     *SyncPeerService
+
+	peerHeap     *peers.PeerHeap
+	subscription blockchain.Subscription
+	id           string // node ID
+}
+
+func NewSyncPeersManager(
+	logger hclog.Logger,
+	network Network,
+	blockchain Blockchain,
+) peers.SyncPeers {
+	return &SyncPeersManager{
+		logger:     logger.Named(SyncPeersManagerLoggerName),
+		network:    network,
+		blockchain: blockchain,
+		server: NewSyncPeerService(
+			logger,
+			blockchain,
+			network,
+		),
```

Comment:
Does this look right to you?

cc @dbrajovic

Reply:
the way to decide best peer depends on consensus, so there is no problem